mypy/types.py

```diff
@@ -2575,6 +2576,17 @@ def flatten_nested_unions(types: Iterable[Type],
     return flat_items
 
 
+def reduce_proper_noreturn_types(types: Iterable[Type]) -> List[Type]:
+    """Remove one or multiple NoReturn types if there is at least one other type.
+
+    Assumes the union was flattened (except for possible union type aliases). The
+    implementation CANNOT handle type aliases, because they might not yet be expanded.
+    """
+    filtered: List[Type] = [tp for tp in types if not is_proper_noreturn_type(tp)]
+    # If nothing is left, we know there was at least one NoReturn.
```

Comment:
Or we can have an empty `types[]` iterable.

Reply:
I assumed this cannot happen, because in the call context of this method (constructor of Union) there must be at least one item in the Union.

So are you fine if I just add an `assert len(types) >= 1` and chaning the type of `types` to `List` (instead of `Iterable`)

Reply:
It can be `Sequence`, the same as `__init__` has